base/resources/scripts/languagecheck.js

```diff
@@ -0,0 +1,40 @@
+/* Nodejs to set up a server on 8081 to modify the argv.json file
+   used by in-browser VSCode. Uses the accept-language header.
+   Also used by jupyter to determine language.
+ */
+var http = require("http");
+var fs = require("fs");
+
+http
+  .createServer(function (req, res) {
+    res.writeHead(200, { "Content-Type": "text/html" });
+    var lang = req.headers["accept-language"];
+    lang ? lang : (lang = "en");
+    lang = lang.substring(0, 2);
+
+    lang === "fr" || lang === "en" ? lang : (lang = "en");
+
+    // READ THE FILE AND COMPARE TO LANG IF IT CHANGES DO THE FOLLOWING, SYNC to not write to it as well
+    var file = fs.readFileSync(
+      "/home/jovyan/.config/Code/User/argv.json",
+      "utf-8"
+    ); // must specify encoding to return string
+    var browserLang = '"locale":"' + lang + '"'; //search using this, "locale":"en" if browserLang not in file then change
+
+    if (file.indexOf(browserLang) == -1) {
+      // if the file index of browserLang == -1 (meaning not found) then write to file
+      fs.writeFile(
```

Comment:
Is there a reason for going with a synchronous read, but an async write?

Reply:
My (possibly mis)-understanding was that the synchronous read would prevent a throwing of an error on that line 24 where it would try and retrieve the browserlang before the operation was complete.